Issue / 7876 Reset Search Console Gathering Data to 48 Hours

10upsimon opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- Fixes #7876

## Relevant technical choices

- Added new `gatheringDataWaitTime` prop to `gatheringDataWaitTime` component.
- Moved calculation logic of wait time to `assets/js/components/notifications/ZeroDataStateNotifications/index.js`.
- Validated that VRT tests still pass.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

hussain-t commented on assets/js/components/notifications/ZeroDataStateNotifications/GatheringDataNotification.js:
```diff
@@ -45,31 +48,32 @@ export default function GatheringDataNotification( { title } ) {
-			onDismiss={ handleOnDismiss }
-			isDismissible
-		/>
+		gatheringDataWaitTime && (
```
Instead of conditionally rendering `BannerNotification` with the `&&` operator, can we return `null` early if `gatheringDataWaitTime` isn't available like so:

```js
if ( ! gatheringDataWaitTime ) {
	return null;
}
```

10upsimon replied:
> Instead of conditionally rendering `BannerNotification` with the `&&` operator, can we return `null` early if `gatheringDataWaitTime` isn't available like so:
> 
> ```js
> if ( ! gatheringDataWaitTime ) {
> 	return null;
> }
> ```

@hussain-t I think that is a fair observation, I will make said change.

hussain-t commented on assets/js/components/notifications/ZeroDataStateNotifications/index.js:
```diff
@@ -134,25 +134,34 @@ export default function ZeroDataStateNotifications() {
 	}
 
 	let gatheringDataTitle;
+	let gatheringDataWaitTime;
```
Can we simplify it by setting the default value as `72`?
```suggestion
	let gatheringDataWaitTime = '72';
```

hussain-t commented on assets/js/components/notifications/ZeroDataStateNotifications/index.js:
```diff
@@ -134,25 +134,34 @@ export default function ZeroDataStateNotifications() {
 	let gatheringDataTitle;
+	let gatheringDataWaitTime;
 	if ( analyticsGatheringData && searchConsoleGatheringData ) {
+		gatheringDataWaitTime = '72';
```
By setting the default value, we don't need it here.
```suggestion
```

hussain-t commented on assets/js/components/notifications/ZeroDataStateNotifications/index.js:
```diff
@@ -134,25 +134,34 @@ export default function ZeroDataStateNotifications() {
 			'google-site-kit'
 		);
 	} else if ( analyticsGatheringData ) {
+		gatheringDataWaitTime = '72';
```
We don't need it here as well.
```suggestion
```

hussain-t commented on assets/js/components/notifications/ZeroDataStateNotifications/index.js:
```diff
@@ -134,25 +134,34 @@ export default function ZeroDataStateNotifications() {
 			'google-site-kit'
 		);
 	} else if ( searchConsoleGatheringData ) {
+		gatheringDataWaitTime = '48';
```
We just need to set it here.

hussain-t reviewed (changes requested):
Thanks, @10upsimon. This looks good. However, I've left a few nit-picky comments for simplicity and better readability. Please take a look and LMK WDYT.

10upsimon commented:
> Thanks, @10upsimon. This looks good. However, I've left a few nit-picky comments for simplicity and better readability. Please take a look and LMK WDYT.

@hussain-t thank you, I've addressed all of your feedback, much appreciated.

hussain-t reviewed (approved):
Thanks, @10upsimon. LGTM 👍 

Note: VRT failures seem unrelated to the changes in this PR.

tofumatt commented on assets/js/components/notifications/ZeroDataStateNotifications/GatheringDataNotification.js:
```diff
@@ -35,7 +35,10 @@ import GatheringDataIcon from '../../../../svg/graphics/zero-state-red.svg';
-export default function GatheringDataNotification( { title } ) {
+export default function GatheringDataNotification( {
+	title,
+	gatheringDataWaitTime,
```
```suggestion
	gatheringDataWaitTime,
```

It's a bit odd that we've left this as a string, and also that we aren't using `_n()` for translations. I know in this instance we don't expect non-plural amounts, but we should be consistent with our translation approaches and use:

1. numbers not strings for numerical values
2. `_n()` to account for different values in different locales

I know @10upsimon is out today so I'll make this minor change then get this merged.
